File: Rust/Easy/Accepted/kth-missing-positive-number_3.rs
impl Solution {
    pub fn find_kth_positive(arr: Vec<i32>, k: i32) -> i32 {
        
        let mut i = 0;
        let mut j = 1;
        let mut k = k;
        while i < arr.len() {
            if arr[i] == j {
                i += 1;
            } else {
                k -= 1;
                if k == 0 {
                    return j;
                }
            }
            j += 1;
        }
        j + k - 1
    }
}

File: Java/Easy/Accepted/kth-missing-positive-number_3.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        
        int n = arr.length;
        int left = 0, right = n;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (arr[mid] - (mid + 1) >= k) {
                right = mid;
            } else {
                left = mid + 1;
            }
        }
        return left + k;
    }
}